src/components/sync-blocking-checker.ts

```diff
@@ -0,0 +1,54 @@
+import { kInsertions, kRemovals, Lyra, OpaqueDocumentStore, OpaqueIndex, Schema } from "../types.js";
+
+// Web platforms don't have process. React-Native doesn't have process.emitWarning.
+const warn =
+  globalThis.process?.emitWarning ??
+  function emitWarning(message: string, options: { code: string }) {
+    console.warn(`[WARNING] [${options.code}] ${message}`);
+  };
+
+export function trackInsertion<S extends Schema, I extends OpaqueIndex, D extends OpaqueDocumentStore>(
```

Comment:
Now that all the methods are async, do we still need this checker at all?

Reply:
Well, the method are marked as async, but if there is no await inside they behave like sync and therefore block the event loop. So we still do need this, unfortunately.